addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/handler/NibeHeatPumpHandler.java

```diff
@@ -0,0 +1,544 @@
+import org.openhab.binding.nibeheatpump.internal.models.VariableInformation.Type;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NibeHeatPumpHandler} is responsible for handling commands, which
+ * are sent to one of the channels.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class NibeHeatPumpHandler extends BaseThingHandler implements NibeHeatPumpEventListener {
+
+    private Logger logger = LoggerFactory.getLogger(NibeHeatPumpHandler.class);
+
+    private PumpModel pumpModel;
+    private NibeHeatPumpConfiguration configuration;
+
+    private NibeHeatPumpConnector connector;
+    private int timeout = 4500;
+
+    private boolean reconnectionRequest;
+
+    private NibeHeatPumpCommandResult writeResult;
+    private NibeHeatPumpCommandResult readResult;
+
+    private ScheduledFuture<?> connectorTask;
+    private ScheduledFuture<?> pollingJob;
+    private List<Integer> itemsToPoll = new ArrayList<Integer>();
+    private List<Integer> itemsToEnableWrite = new ArrayList<Integer>();
+
+    private Map<Integer, CacheObject> stateMap = Collections.synchronizedMap(new HashMap<Integer, CacheObject>());
+
+    protected class CacheObject {
+
+        /** Time when cache object updated in milliseconds */
+        long lastUpdateTime;
+
+        /** Cache value */
+        Double value;
+
+        /**
+         * Initialize cache object.
+         *
+         * @param lastUpdateTimeMs
+         *            Time in milliseconds.
+         *
+         * @param value
+         *            Cache value.
+         */
+        CacheObject(long lastUpdateTime, Double value) {
+            this.lastUpdateTime = lastUpdateTime;
+            this.value = value;
+        }
+    }
+
+    public NibeHeatPumpHandler(Thing thing, PumpModel pumpModel) {
+        super(thing);
+        this.pumpModel = pumpModel;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        int coilAddress = parseCoilAddressFromChannelUID(channelUID);
+
+        if (command.equals(RefreshType.REFRESH)) {
+            logger.debug("Clearing cache value for channel '{}' to refresh channel data", channelUID);
+            clearCache(coilAddress);
+            return;
+        }
+
+        if (!configuration.enableWriteCommands) {
+            logger.info(
+                    "All write commands denied, ignoring command! Change Nibe heat pump binding configuration if you want to enable write commands.");
+            return;
+        }
+
+        if (!itemsToEnableWrite.contains(coilAddress)) {
+            logger.info(
+                    "Write commands to register '{}' not allowed, ignoring command! Add this register to Nibe heat pump binding configuration if you want to enable write commands.",
+                    coilAddress);
+            return;
+        }
+
+        if (connector != null) {
+
```

Comment:
remove all the superfluous empty lines in the following code

Reply:
Done